### ISSUE TITLE
[AWS::SSM::Association] - [BUG] - cfn-lint doesn't recognize `Values: expected type: JSONArray, found: String`

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#3223]
### CloudFormation Lint Version

cfn-lint 0.87.1

### What operating system are you using?

Windows 11

### Describe the bug

cfn-lint doesn't recognize the following error message:

```
Resource handler returned message:
"Model validation failed (#/Targets/0/Values: expected type: JSONArray, found: String)"
(RequestToken: REQUESTTOKEN, HandlerErrorCode: InvalidRequest)
```

for the following example:

```yaml
  SSMAssociation:
    Type: AWS::SSM::Association
    Properties:
      Name: "SSM Document Name"
      ScheduleExpression: rate(2 days)
      Targets:
        - Key: InstanceIds
          Values: !GetAtt EC2Instance.InstanceId
```

This is how it should look like:

```yaml
  SSMAssociation:
    Type: AWS::SSM::Association
    Properties:
      Name: "SSM Document Name"
      ScheduleExpression: rate(2 days)
      Targets:
        - Key: InstanceIds
          Values:
            - !GetAtt EC2Instance.InstanceId
```

### Expected behavior

cfn-lint finds this error.

### Reproduction template

See above.

**Comment by kddejong** [comment]
Hmm, should be able to find this one.  I wonder if we aren't handling JSON values well.  I'll look into this.

**Comment by kddejong** [comment]
Already fixed in v1 but is a bug in v0.
